CDS-96727 - KB on creating a cascading field when creating a JIRA issue using APIs (#6782)
harness/developer-hub: kb/continuous-delivery/articles/create-cascading-fields-jira.md

---

> title: Create cascading fields for JIRA issues using the HTTP step

Author: ```suggestion
title: Use the HTTP step to create the unsupported fields when creating Jira issues
```

> description: Instructions on setting cascading fields when creating a JIRA issue using the HTTP step.

Author: ```suggestion
description: Instructions on setting unsupported fields when creating a JIRA issue using the HTTP step.
```

---

> This article talks about setting a [cascading list](https://confluence.atlassian.com/jirakb/creating-dependent-cascading-lists-in-jira-server-datacenter-1142426572.html) when creating a JIRA issue. By default, Jira plugins don't support cascading lists. You can configure cascading lists for your Jira issues using an [HTTP step](/docs/continuous-delivery/x-platform-cd-features/cd-steps/utilities/http-step/). 

Author: ```suggestion
This article talks about using the HTTP step to set the fields that are not supported by default by the Jira plugin using an example. 

We will discuss how to set a [cascading list](https://confluence.atlassian.com/jirakb/creating-dependent-cascading-lists-in-jira-server-datacenter-1142426572.html) using the HTTP step. By default, Jira plugins don't support cascading lists. You can configure cascading lists for your Jira issues using an [HTTP step](/docs/continuous-delivery/x-platform-cd-features/cd-steps/utilities/http-step/). 
```

---
title: Use the HTTP step to set unsupported fields when creating Jira issues
description: Instructions on setting unsupported fields when creating a JIRA issue using the HTTP step.
---

This article talks about using the HTTP step to set the fields that are not supported by default by the Jira plugin using an example. 

> We will discuss how to set a [cascading list](https://confluence.atlassian.com/jirakb/creating-dependent-cascading-lists-in-jira-server-datacenter-1142426572.html) using the HTTP step. By default, Jira plugins don't support cascading lists. You can configure cascading lists for your Jira issues using an [HTTP step](/docs/continuous-delivery/x-platform-cd-features/cd-steps/utilities/http-step/). 

Author: ```suggestion
We will discuss how to set a [cascading list](https://confluence.atlassian.com/jirakb/creating-dependent-cascading-lists-in-jira-server-datacenter-1142426572.html) using the HTTP step. By default, Jira plugins don't support cascading lists and other custom types. For the list of supported Jira fields, go to [Add issue fields](/docs/continuous-delivery/x-platform-cd-features/cd-steps/ticketing-systems/create-jira-issues-in-cd-stages/#add-issue-fields).

You can configure cascading lists for your Jira issues using an [HTTP step](/docs/continuous-delivery/x-platform-cd-features/cd-steps/utilities/http-step/). 
```

---
title: Create cascading fields for JIRA issues using the HTTP step
description: Instructions on setting cascading fields when creating a JIRA issue using the HTTP step.
---

This article talks about setting a [cascading list](https://confluence.atlassian.com/jirakb/creating-dependent-cascading-lists-in-jira-server-datacenter-1142426572.html) when creating a JIRA issue. By default, Harness doesn't support cascading lists. You can configure the cascading list for your Jira issues using an [HTTP step](/docs/continuous-delivery/x-platform-cd-features/cd-steps/utilities/http-step/). 

Create a pipeline with a [Custom stage](/docs/platform/pipelines/add-a-stage/#add-a-custom-stage) and configure the following steps under the Execution tab of the pipeline. 

## Configure the HTTP step

In the HTTP step's **Step Parameters**, configure the following fields.

- URL: Enter your Jira server details. For example, `https://<your_JIRA_domain>/rest/api/2/issue`.
- Method: Select **POST** HTTP method.
- Request Body: Enter the cURL command to create cascading field. Here's a sample:  

  ```

>   {

Reviewer: @roshnisarangadharan can the payload be mentioned in kind of code format

Author: I've followed the UI step parameter steps for HTTP step. Have added a pipeline YAML later in this page to make sure users don't get confused.

  "fields": {
    "project": { 
      "key": "TES"
     },
    "summary": "Testing select list (cascading)",
    "issuetype": {
      "id": "10003"
     },
    "customfield_10500": {
      "value": "Parent 1",
      "child": {
        "value": "P1 Child 1"
      }      
    }
   }
  }
  ```
- Optional Configuration: Enter the following optional configurations.
  - Headers: Enter the following Key:Value information:
    - Content-Type: `application/json`

>     - Authorization: "Basic `<authorization_key_to_your_Jira_server>`"

Author: ```suggestion
    - Authorization: `Basic <authorization_key_to_your_Jira_server>`
```

>   - Headers: Enter the following Key:Value information:
>     - Content-Type: application/json
>     - Authorization: Basic `<authorization_key_to_your_Jira_server>`
>   - Output: Enter the variable name and value to get the Jira Issue Id from the pipeline output.  For example:
>     - Variable Name: issue_key
>     - Value: Choose Expression input type and enter `<+json.select("key", httpResponseBody)>` 

Reviewer 2: ```suggestion
  - Headers: Enter the following Key:Value information:
    - Content-Type: `application/json`
    - Authorization: "Basic `<authorization_key_to_your_Jira_server>`"
  - Output: Enter the variable name and value to get the Jira Issue Id from the pipeline output.  For example:
    - Variable Name: `<issue_key>`
    - Value: Choose Expression input type and enter `<+json.select("key", httpResponseBody)>` 
```

Reviewer 2: This one really is just a suggestion, consider putting the `values` in quotes or back tics to indicate they are or are not replaced with something. 

I think Issue key should definitely be `<issue_key>` though
